### PR TITLE
Show number of days and players on Commons Overview

### LINKED ISSUE #28: FEATURE: Add duration and total players tracking
The user commons page should show how many days it has been since the game started and how many players there are.

Part of the larger issue #11

### PR DESCRIPTION
**Pull request by mfletcher2**
This PR adds the display of the number of days since the start and the number of players. The day count is computed with the start date and the current date on the client with the new function `daysSinceTimestamp` in `dateUtils`. The number of players is retrieved from a new endpoint to get a single `CommonsPlus`. It contains information about the number of players and cows in addition to the other `Commons` fields. For testing purposes, the player delete endpoint now returns a message that says the new number of players.

# Screenshots
Before:
Screenshot_20230531_231445
After:
Screenshot_20230531_231512

# Storybook
[Before](https://ucsb-cs156-s23.github.io/proj-happycows-s23-6pm-1/storybook/?path=/story/components-commons-commonsoverview--uncontrolled)
[After](https://ucsb-cs156-s23.github.io/proj-happycows-s23-6pm-1/prs/38/storybook/?path=/story/components-commons-commonsoverview--uncontrolled)
(The counters appear to be working on the before storybook because they were given values in the fixtures, but not implemented in the rest of the code)

Closes #28

### CODE REVIEW

**Comment by GlassBricks** [comment]
The handling of the day number is good.

However, I think the number of players should be computed (by the backend) instead of a manually edited field in `Commons`. It definetely doesn't make sense to have it in `CreateCommonsParams`.
Instead, there's already a backend thing that includes this computed info (`CommonsPlus`, and the endpoint `commons/allplus`); we could add another endpoint to get info for a specific commons, and use that in the frontend.

**Review by GlassBricks — CHANGES_REQUESTED** [review]
The handling of the day number is good.

However, I think the number of players should be computed (by the backend) instead of a manually edited field in `Commons`. It definitely doesn't make sense to have it in `CreateCommonsParams`.
There's already a backend thing that includes this computed info (`CommonsPlus`, and the endpoint `commons/allplus`); we could add another endpoint to get info for a specific commons, and use that in the frontend.

File: frontend/src/main/pages/PlayPage.js [hunk 1]
```diff
@@ -47,6 +47,20 @@ export default function PlayPage() {
     );
```
Comment (GlassBricks) [inline frontend/src/main/pages/PlayPage.js]: We don't need another backend GET for normal commons; you can use `commonsPlus.commons` instead, avoiding an extra query

**Review by GlassBricks — CHANGES_REQUESTED** [review]

File: frontend/src/main/pages/PlayPage.js [hunk 2]
```diff
@@ -143,7 +128,7 @@ export default function PlayPage() {
           <br />
           {!!userCommons &&
             <CardGroup >
-              <ManageCows userCommons={userCommons} commons={commons} onBuy={onBuy} onSell={onSell} />
+              <ManageCows userCommons={userCommons} onBuy={onBuy} onSell={onSell} />
```
Comment (GlassBricks) [inline frontend/src/main/pages/PlayPage.js]: I think you still need `commons={commonsPlus.commons}` here... now market cow price isn't being shown

**Review by GlassBricks — CHANGES_REQUESTED** [review]

File: frontend/src/main/pages/PlayPage.js (hunk 2 above)
Comment (mfletcher2) [reply frontend/src/main/pages/PlayPage.js]: Thanks for catching that

**Review by GlassBricks — DISMISSED** [review]
LGTM

File: frontend/src/main/pages/PlayPage.js [hunk 3]
```diff
@@ -32,20 +32,19 @@ export default function PlayPage() {
...
         method: "GET",
-        url: "/api/commons",
+        url: "/api/commons/plus",
         params: {
           id: commonsId
         }
       }
     );
-  // Stryker enable all 
+  // Stryker enable all
```
Comment (pconrad) [inline frontend/src/main/pages/PlayPage.js]: Should be 
```suggestion
  // Stryker restore all
```

Please see: https://ucsb-cs156-s23.slack.com/archives/C050779M19P/p1685726928710669

File: frontend/src/main/utils/dateUtils.js [hunk 4]
```diff
@@ -6,4 +6,15 @@ const timestampToDate = (timestamp) => {
     return (date.getFullYear() + "-" + (padWithZero(date.getMonth()+1)) + "-" + padWithZero(date.getDate()));
 }
 
-export {timestampToDate, padWithZero};
\ No newline at end of file
+// Disabling stryker here because the returned value depends on the date the test is run
+// so a deterministic test is not possible
+// Stryker disable all
```
Comment (pconrad) [inline frontend/src/main/utils/dateUtils.js]: There's a better approach to this: mock the system time:

```js
jest.useFakeTimers().setSystemTime(new Date('2020-01-01'));
```

See also:  https://stackoverflow.com/questions/29719631/how-do-i-set-a-mock-date-in-jest/63377110#63377110

File: frontend/src/main/utils/dateUtils.js [hunk 5]
```diff
@@ -6,4 +6,15 @@ const timestampToDate = (timestamp) => {
...
+const daysSinceTimestamp = (date) => {
+    var today = new Date();
+    var startingDate = new Date(date);
+    var timeDiff = Math.abs(today.getTime() - startingDate.getTime());
+    return Math.ceil(timeDiff / (1000 * 3600 * 24));
+}
+// Stryker enable all
```
Comment (pconrad) [inline frontend/src/main/utils/dateUtils.js]: we can probably eliminate the need for a stryker disable here, so we can remove this; but if we don't, it should be

```suggestion
// Stryker restore all
```

**Review by pconrad — CHANGES_REQUESTED** [review]
Good overall, but there are a few things that need to be fixed.

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**